smdebug/tensorflow/keras.py

```diff
@@ -548,36 +549,56 @@ def _save_metrics(self, batch, logs, force_save=False):
     def _save_layer_input_and_outputs(self):
         if is_tf_version_2x() is False:
             return
+        layer_collection = (
+            {self.get_collection(CollectionKeys.LAYERS)}
+            if self._is_collection_being_saved_for_step(CollectionKeys.LAYERS)
+            else set()
+        )
         for layer_name in self.saved_layers:
             # Save Input
-            tensor = self.saved_layers[layer_name].layer_input
-            export_name = get_export_name_for_keras(layer_name, tensor_type="input", tensor=tensor)
-            input_collection = (
-                {self.get_collection(CollectionKeys.LAYERS)}
-                if self._is_collection_being_saved_for_step(CollectionKeys.LAYERS)
-                else set()
-            )
-            t = tensor[0] if isinstance(tensor, list) and len(tensor) else tensor
-            if hasattr(t, "numpy") is False:
-                self.logger.warning("cannot save layer values during forward pass with tf.function")
-                continue
-            else:
-                self._save_tensor_to_file(export_name, tensor, input_collection)
-
+            layer_inputs = self.saved_layers[layer_name].layer_input
+            for layer_idx, tensor in enumerate(layer_inputs):
+                if isinstance(tensor, list):
+                    tensor_list = tensor
+                else:
+                    tensor_list = [tensor]
+                if hasattr(tensor_list[0], "numpy") is False:
+                    self.logger.warning(
+                        "cannot save layer values during forward pass with tf.function"
+                    )
+                    continue
+                else:
+                    for t_idx, t in enumerate(tensor_list):
+                        export_name = get_export_name_for_keras(
+                            layer_name,
+                            tensor_type="input",
+                            tensor=tensor,
+                            layer_idx=layer_idx,
+                            tensor_idx=t_idx,
+                        )
+                        self._save_tensor_to_file(export_name, t, layer_collection)
```

Comment:
this looks similar to what's done for outputs below. possible to make it common?